src/Zigzag_persistence/include/gudhi/filtered_zigzag_persistence.h

```diff
@@ -0,0 +1,491 @@
+  void insert_face(face_key faceID,
+                   const BoundaryRange& boundary,
+                   dimension_type dimension,
+                   filtration_value filtrationValue) {
+    ++numArrow_;
+
+    [[maybe_unused]] auto res = handleToKey_.try_emplace(faceID, numArrow_);
+
+    GUDHI_CHECK(res.second, "Zigzag_persistence::insert_face - face already in the complex");
+
+    keyToFiltrationValue_.try_emplace(numArrow_, filtrationValue);
+
+    // Compute the keys of the faces of the boundary.
+    std::set<internal_key> translatedBoundary;  // set maintains the natural order on indices
+    for (auto b : boundary) {
+      translatedBoundary.insert(handleToKey_.at(b));  // TODO: add possibilities of coefficients
+    }
+
+    pers_.insert_face(translatedBoundary, dimension);
+  }
+
+  /**
+   * @brief Updates the zigzag persistence diagram after the removal of the given face.
+   *
+   * @param faceID ID representing the face to remove. Should be the same than the one used to insert it.
+   * @param dimension Dimension of the face.
+   * @param filtrationValue Filtration value associated to the removal.
+   * Assumed to be always larger or equal to previously used filtration values or always smaller or equal than previous
+   * values, ie. the changes are monotonous.
+   */
+  void remove_face(face_key faceID, dimension_type dimension, filtration_value filtrationValue) {
+    ++numArrow_;
+
+    auto it = handleToKey_.find(faceID);
+    GUDHI_CHECK(it != handleToKey_.end(), "Zigzag_persistence::remove_face - face not in the complex");
+
+    keyToFiltrationValue_.try_emplace(numArrow_, filtrationValue);
+
+    pers_.remove_face(it->second, dimension);
+    handleToKey_.erase(it);
+  }
+
+  /**
+   * @brief To use when a face is neither inserted nor removed, but the filtration moves along the identity operator
+   * on homology level. Useful to keep the birth/death indices aligned when insertions/removals are purposely skipped
+   * to avoid useless computation.
+   */
+  void apply_identity() {
+    ++numArrow_;
+    pers_.apply_identity();
+  }
+
+  /**
+   * @brief Outputs through the given callback method all current infinite bars.
+   * 
+   * @tparam F Type of the callback method. Takes two arguments: the dimension of the cycle and the birth value
+   * of the cycle.
+   * @param stream_infinite_interval Method processing the unpaired birth values.
+   */
+  template <typename F>
+  void get_current_infinite_intervals(F&& stream_infinite_interval) {
+    pers_.get_current_infinite_intervals(
+        [&](dimension_type dim, internal_key birth) { stream_infinite_interval(dim, keyToFiltrationValue_.at(birth)); });
+  }
+
+ private:
+  template <typename key_type, typename value_type>
+  using dictionary = std::unordered_map<key_type, value_type>;  // TODO: benchmark with other map types
+
+  dictionary<face_key, internal_key> handleToKey_;                   /**< Map from input keys to internal keys. */
```

Comment:
Trying to think of a specific case.
Imagine I am using a Simplex_tree to represent a complex, and I use `Simplex_handle` as `face_key`. To map handles to keys, I am tempted to store the key in the key field of the simplex tree nodes instead of using a separate map. So I should stick to the basic Zigzag_persistence class, right?

Reply:
Yes, I think so. From the keys in the returned diagram, it will also be easy to retrieve the filtration values, so the wrappers are not really necessary. It is probably what I will do in the oscillating rips class.